evm/src/cpu/kernel/interpreter.rs

```diff
@@ -1078,56 +1172,95 @@ impl<'a> Interpreter<'a> {
             + (U256::from(self.generation_state.registers.gas_used) << 192);
         self.generation_state.registers.program_counter = new_program_counter;
 
+        if opcode == 244 {
+            println!(
+                "new_pc = {new_program_counter}, new_label = {:?}, sys_calldatasize = {:?}, sys_gasprice = {:?},",
+                KERNEL.offset_name(new_program_counter),
+                KERNEL.global_labels["sys_calldatasize"],
+                KERNEL.global_labels["sys_gasprice"],
+
+            );
+        }
+
         self.set_is_kernel(true);
         self.generation_state.registers.gas_used = 0;
         self.push(syscall_info)
     }
 
-    fn set_jumpdest_bit(&mut self, x: U256) -> U256 {
+    fn get_jumpdest_bit(&self, offset: usize) -> U256 {
         if self.generation_state.memory.contexts[self.context()].segments
             [Segment::JumpdestBits.unscale()]
         .content
         .len()
-            > x.low_u32() as usize
+            > offset
         {
             self.generation_state.memory.get(MemoryAddress {
                 context: self.context(),
                 segment: Segment::JumpdestBits.unscale(),
-                virt: x.low_u32() as usize,
+                virt: offset,
             })
         } else {
             0.into()
         }
     }
-    fn run_jump(&mut self) -> anyhow::Result<(), ProgramError> {
-        let x = self.pop()?;
 
-        let jumpdest_bit = self.set_jumpdest_bit(x);
+    pub(crate) fn get_jumpdest_bits(&self, context: usize) -> Vec<bool> {
+        self.generation_state.memory.contexts[context].segments[Segment::JumpdestBits.unscale()]
+            .content
+            .iter()
+            .map(|x| x.bit(0))
+            .collect()
```

Comment:
Is this a hot function? I _think_ there are tricks to make this very fast and I do not think `bit()` from `uint` is optimal.

Reply:
contract bytecode can be quite long, so we may want to have this optimized, to some extent. But I don't think that will be _the_  bottleneck anyway

Reply:
In the end this function is only used for testing that `jumpdest_analysis` correctly write the jumpdest bits.